agent/rpi_helper.py

```diff
@@ -144,3 +145,43 @@ def auto_upgrades_enabled():
             '${distro_id}:${distro_codename}-security' in allowed_origins
     else:
         return False
+
+
+def kernel_cmdline():
+    """
+    Parses kernel parameters (aka cmdline).
+    :return: A dict where 'name' is kernel parameter name and 'value' is its value or empty string if no value provided.
+    """
+    cmdline_path = Path('/proc/cmdline')
+    cmdline_matches = re.compile(r"([\w\-\.]+)(\=(\"[\w\W]+\"|[\w\S]+)?)?").findall(cmdline_path.read_text())
+    return {name: value.strip('"') for name, _, value in cmdline_matches}
+
+
+def kernel_deb_package():
+    """
+    Finds which currently installed deb package contains the currently running kernel.
+    :return:
+        If the currently running kernel was installed by deb package:
+            a dict where 'source_name' and 'source_version' are the same as returned by get_deb_packages().
+        Otherwise: None
+    """
+    import apt
+
+    boot_image = kernel_cmdline().get('BOOT_IMAGE')
```

Comment:
My RPI3 default installation doesn't have the 'BOOT_IMAGE' kernel command line parameter. what does that mean?

Reply:
It means that it boots differently, not with a vmlinuz image, I wrote about it [here](https://github.com/WoTTsecurity/api/issues/482#issuecomment-546829943). Pi is not vulnerable anyway, since it's ARM.